### PR TITLE
[feature]GLM4.5 support gatingtopk

### PR DESCRIPTION
**Pull request by 1092626063**
### What this PR does / why we need it?

Past：npu_moe_gating_top_k can only support 'group_count=256' pattern
Now：npu_moe_gating_top_k support all size of group_count

So，GLM4.5 can use this fusion operator

### CODE REVIEW

File: vllm_ascend/ops/moe/experts_selector.py [hunk 1]
```diff
@@ -183,29 +183,25 @@ def _select_experts_with_fusion_ops(
...
 
     topk_weights, topk_ids, row_idx = None, None, None
-    # NOTE: now npu_moe_gating_top_k can only support 'group_count=256' pattern
-    is_deepseek_v3_r1 = global_num_experts == 256
-    if is_deepseek_v3_r1:
+    norm_type = 0 if scoring_func == "softmax" else 1
+    if custom_routing_function is None:
+        if e_score_correction_bias is not None and \
+            e_score_correction_bias.dtype != router_logits.dtype:
+            e_score_correction_bias = e_score_correction_bias.to(router_logits.dtype)
```
Comment (whx-sjtu) [inline vllm_ascend/ops/moe/experts_selector.py]: To avoid performing to_dtype per-layer, you should do to_dtype to `self.e_score_correction_bias` of `AscendFusedMoe`.

File: vllm_ascend/ops/moe/experts_selector.py [hunk 2]
```diff
@@ -183,29 +183,25 @@ def _select_experts_with_fusion_ops(
...
             renorm=0,  # 0: softmax->topk(fix); 1: topk->softmax
-            norm_type=1,  # 0: softmax; 1: sigmoid(fix)
+            norm_type=norm_type,  # 0: softmax; 1: sigmoid
             # out_flag=False, # todo new api; should the third output be output
             # y2_flag=False, # old api; should the third output be output
             routed_scaling_factor=1,
             eps=float(1e-20))
         row_idx = return_row_idx(hidden_states, top_k)
-    if not use_grouped_topk and custom_routing_function is None and scoring_func == "softmax":
-        topk_weights, topk_ids, row_idx = torch_npu.npu_moe_gating_top_k_softmax(
```
Comment (whx-sjtu) [inline vllm_ascend/ops/moe/experts_selector.py]: Is it certain that the functionality of `torch_npu.npu_moe_gating_top_k_softmax` are included in `torch_npu.npu_moe_gating_top_k`?

Comment (1092626063) [reply vllm_ascend/ops/moe/experts_selector.py]: yes

File: vllm_ascend/ops/fused_moe/experts_selector.py [hunk 3]
```diff
@@ -183,29 +183,26 @@ def _select_experts_with_fusion_ops(
...
             1,  # 0: the maximum in the group; 1: topk2.sum(fix)
             renorm=0,  # 0: softmax->topk(fix); 1: topk->softmax
-            norm_type=1,  # 0: softmax; 1: sigmoid(fix)
+            norm_type=norm_type,  # 0: softmax; 1: sigmoid
             # out_flag=False, # todo new api; should the third output be output
             # y2_flag=False, # old api; should the third output be output
             routed_scaling_factor=1,
             eps=float(1e-20))
         row_idx = return_row_idx(hidden_states, top_k)
-    if not use_grouped_topk and custom_routing_function is None and scoring_func == "softmax":
```
Comment (momo609) [inline vllm_ascend/ops/fused_moe/experts_selector.py:198]: Provide pre- and post-replacement performance data to prevent degradation.

File: vllm_ascend/ops/moe/experts_selector.py [hunk 4]
```diff
@@ -94,8 +94,8 @@ def select_experts(hidden_states: torch.Tensor,
             e_score_correction_bias=e_score_correction_bias,
             global_num_experts=global_num_experts,
         )
-    if row_idx is None:
-        row_idx = return_row_idx(hidden_states, top_k)
+        if row_idx is None:
```
Comment (weijinqian0) [inline vllm_ascend/ops/moe/experts_selector.py]: You can delete this code.

File: vllm_ascend/ops/fused_moe/experts_selector.py [hunk 5]
```diff
@@ -171,34 +168,34 @@ def _select_experts_with_fusion_ops(
...
+    if scoring_func == "softmax":
+        norm_type = 0
+        topk_group = 1
+        num_expert_group = 1
+    else:
+        norm_type = 1
+    if e_score_correction_bias is not None and \
+        e_score_correction_bias.dtype != router_logits.dtype:
+        e_score_correction_bias = e_score_correction_bias.to(
+            router_logits.dtype)
```
Comment (whx-sjtu) [inline vllm_ascend/ops/fused_moe/experts_selector.py:184]: The necessity here is that GLM4 uses fp32 for gate. So we have to do this dtype transform here.

**Review by whx-sjtu — APPROVED** [review]
LGTM

File: vllm_ascend/ops/moe/experts_selector.py (hunk 1 above)
Comment (1092626063) [reply vllm_ascend/ops/moe/experts_selector.py]: Note TODO here: GLM4.5 is special, it use float32 to calculate, but in `AscendFusedMoE::init()` `e_score_correction_bias` changed dtype from float32 to bfloat16, this is different from deepseek and qwen models. So the logits in `AscendFusedMoE::init()` needs to reconsider later.

File: vllm_ascend/ops/fused_moe/experts_selector.py (hunk 3 above)
Comment (1092626063) [reply vllm_ascend/ops/fused_moe/experts_selector.py:198]: OK，Performance：
GLM4.5-w8a8, TPS improve 6%
Qwen3, the same as before
